findspam.py

```diff
@@ -316,6 +321,20 @@ def bad_pattern_in_url(s, site, *args):
         return False, ""
 
 
+def bad_ns_for_url_domain(s, site, *args):
+    for domain in set([get_domain(link) for link in post_links(s)]):
+        try:
+            ns = dns.resolver.query(domain, 'ns')
+        except dns.exception.DNSException as exc:
+            # TODO: log exception?
+            continue
+        nameservers = [server.target.to_text() for server in ns]
+        if any([ns.endswith('.namecheaphosting.com.') for ns in nameservers]):
```

Comment:
I completely agree; but let's see if this is feasible at all before generalizing it, shall we? Devising a extension mechanism would complicate this proof of concept, potentially needlessly if it turns out to be too heavy and/or FP-prone. I can't think of another provider that we could realistically blacklist at this point, anyway.